network/api/client.go

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright (C) 2020. Nuts community
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+package api
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/nuts-foundation/nuts-node/network/rfc004"
+	"io/ioutil"
+	"net/http"
+	"strings"
+	"time"
+)
+
+// HttpClient holds the server address and other basic settings for the http client
+type HttpClient struct {
+	ServerAddress string
+	Timeout       time.Duration
+}
+
+func (hb HttpClient) CreateDocument(payloadType string, payload []byte, signingKeyID string, attachKey bool, timestamp time.Time, fieldsOpts ...rfc004.FieldOpt) (rfc004.Document, error) {
```

Comment:
The HTTPClient doesn't need to implement the same interface as server side.
Only the CLI will use it and that can use the struct directly.

Reply:
That makes testing harder because you can't mock the struct